### PR TITLE
Added utility functions converting string to ulong

### PR DESCRIPTION
**Pull request by larsewi**
Added functions StringToUlong, StringToUlongDefaultOnError,
StringToUlongExitOnError.

Ticket: None
Changelog: Title
Signed-off-by: Lars Erik Wik <lars.erik.wik@northern.tech>

### CODE REVIEW

File: libutils/string_lib.c [hunk 1]
```diff
@@ -601,6 +602,120 @@ long StringToLongExitOnError(const char *str)
...
+    errno = 0;
+    val = strtoul(str, &endptr, 10);
+
+    if ((errno == ERANGE && val == ULONG_MAX))
+    {
+        return ERANGE; // Overflow
+    }
+
+    // Negative numbers should cause underflow
+    if (val > 0) // -0 is ok
```
Comment (olehermanse) [inline libutils/string_lib.c:638]:
```suggestion
    if (val > 0) // "-0" is ok
```

Nitpicking, but I suggest making it clear that you are talking about the string `"-0"` here.

File: libutils/string_lib.c [hunk 2]
```diff
@@ -601,6 +602,120 @@ long StringToLongExitOnError(const char *str)
...
+{
+    nt_static_assert(ERANGE != 0);
+    assert(str != NULL);
+    assert(value_out != NULL);
+
+    char *endptr = NULL;
+    unsigned long val;
+
+    errno = 0;
+    val = strtoul(str, &endptr, 10);
```
Comment (olehermanse) [inline libutils/string_lib.c:630]:
```suggestion
    char *endptr = NULL;
    errno = 0;
    const unsigned long val = strtoul(str, &endptr, 10);
```

Declare and assign on the same line;

https://github.com/cfengine/core/blob/master/CONTRIBUTING.md#readability--maintainability

(You can make this change in StringToLong in a separate commit, if it's the same there).

File: libutils/string_lib.c (hunk 1 above)
Comment (larsewi) [reply libutils/string_lib.c]: Agreed

File: libutils/string_lib.c (hunk 2 above)
Comment (larsewi) [reply libutils/string_lib.c]: Good catch

**Review by vpodzime — APPROVED** [review]
ACK

File: libutils/string_lib.c [hunk 3]
```diff
@@ -496,10 +496,9 @@ int StringToLong(const char *str, long *value_out)
     assert(value_out != NULL);
 
     char *endptr = NULL;
-    long val;
 
     errno = 0;
-    val = strtol(str, &endptr, 10);
+    long val = strtol(str, &endptr, 10);
```
Comment (olehermanse) [inline libutils/string_lib.c:501]:
```suggestion
    const long val = strtol(str, &endptr, 10);
```

File: libutils/string_lib.c [hunk 4]
```diff
@@ -601,6 +601,119 @@ long StringToLongExitOnError(const char *str)
...
+int StringToUlong(const char *str, unsigned long *value_out)
+{
+    nt_static_assert(ERANGE != 0);
+    assert(str != NULL);
+    assert(value_out != NULL);
+
+    char *endptr = NULL;
+
+    errno = 0;
+    unsigned long val = strtoul(str, &endptr, 10);
```
Comment (olehermanse) [inline libutils/string_lib.c:628]:
```suggestion
    const unsigned long val = strtoul(str, &endptr, 10);
```

**Review by olehermanse — APPROVED** [review]
